src/Implement/FileOperator.cpp

```diff
@@ -0,0 +1,106 @@
+// Copyright (c) 2021, 鍾淯丞, 周杰仕, 林仁鴻. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include <iostream>
+
+#include "FileOperator.h"
+#include "DownloadStream.h"
+#include "UploadStream.h"
+#include "ResponseHandler.h"
+
+int FileOperator::Upload(const std::unique_ptr<OperateData> &data) {
+    // where to upload include fileName
+    std::string path = data->body["path"];
+    
+    // Binary data that needs to be written
+    std::string binary_data = data->body["data"];
+    
+    if (UploadStream(path).appendToFile(binary_data) == 1)
+        return 200;
+    else
+        return 500;
+}
+
+int FileOperator::Download(const std::unique_ptr<OperateData> &data) {
+    // where to download include fileName
+    std::string path = data->body["path"];
+    
+    // chunk number
+    unsigned long pos = std::stoi(data->body["pos"]);
+    
+    // Binary data from server
+    std::string binary_data = "";
+    
+    if (DownloadStream(path, 2 * 1024 * 1024).ReadChunk(pos, &binary_data) == 1)
+        return 200;
+    else
+        return 500;
+}
+
+void FileOperator::CallingAPI(std::unique_ptr<OperateData> &data) {
+
+    
+    const int client_fd = std::stoi(data->header["client_file_descriptor"]);
```

Comment:
Same as above on error handling.

Reply:
Consider passing the fd in through the response handler for easier unit testing.